source/common/filesystem/posix/filesystem_impl.cc

```diff
@@ -130,12 +130,19 @@ bool InstanceImplPosix::illegalPath(const std::string& path) {
   // instances poking in bad places. We may have to consider conditioning on
   // platform in the future, growing these or relaxing some constraints (e.g.
   // there are valid reasons to go via /proc for file paths).
-  // TODO(htuch): Optimize this as a hash lookup if we grow any further.
-  if (absl::StartsWith(canonical_path.rc_, "/dev") ||
-      absl::StartsWith(canonical_path.rc_, "/sys") ||
-      absl::StartsWith(canonical_path.rc_, "/proc")) {
-    return true;
+  auto paths = []() {
+    CONSTRUCT_ON_FIRST_USE(std::vector<std::string>, {"/dev", "/sys", "/proc"});
```

Comment:
Please just make this a static method on the class.

Reply:
Hmmm, I see the return type of lambda has an issue here, will fix it. But IMO a lambda is better than static class method here, because it's only called here.